extends Node
class_name GameManager

var evidencias: Array = []
var puntuacion := 0
var score_label: Label = null
var log_panel: RichTextLabel = null
var bitacora_texto := ""
var aciertos := 0
var fallos := 0
var total := 0

enum EstadoEvidencia {
	ENCENDIDO,
	APAGADO,
	DESCONECTADO,
	RECOLECTADO
}

func _ready():
	var scene_root = get_tree().get_current_scene()
	print("scene_root:", scene_root.name)
	#score_label = scene_root.get_node_or_null("UIX/InfoPanel/VBoxContainer/ScoreLabel")
	#log_panel = scene_root.get_node_or_null("UIX/InfoPanel/VBoxContainer/ScrollContainer/LogPanel")
	
	#print("score_label:", score_label)
	#print("log_panel:", log_panel)
	
func cambiar_a_scene_selector():
	var selector_scene = preload("res://Scenes/scene_selector.tscn").instantiate()
	get_tree().current_scene.free()
	get_tree().current_scene = selector_scene
	get_tree().root.add_child(selector_scene)
	
func sumar_puntos(valor: int):
	puntuacion += valor
	if score_label:
		score_label.text = str(puntuacion)
		print("Puntaje actualizado en ScoreLabel:", score_label.text)
		print("score_label es:", Game_Manager.score_label, "Tipo:", Game_Manager.score_label.get_class())
	else:
		print("ScoreLabel es null")
	#print("Puntuaci√≥n actual:", puntuacion)

func registrar_accion(texto: String):
	if log_panel:
		log_panel.append_text("üïµÔ∏è " + texto + "\n")
		var entrada := texto + "\n"
		bitacora_texto += entrada
		
func registrar_en_bitacora(mensaje: String) -> void:
	var timestamp = "[" + Time.get_datetime_string_from_system().substr(11, 8) + "]"  # Solo HH:MM:SS
	var entrada = "[color=blue]" + timestamp + "[/color] " + mensaje
	
	if log_panel:
		if log_panel is RichTextLabel:
			print("¬°Es un RichTextLabel!")
		else:
			print("‚ùå No es un RichTextLabel.")
		log_panel.append_text(entrada + "\n")
		log_panel.scroll_to_line(log_panel.get_line_count() - 1)
	else:
		print("‚ùå log_panel no est√° enlazado correctamente.")
		
func guardar_bitacora_en_archivo():
	if not log_panel:
		print("Error: log_panel no encontrado.")
		return

	var ruta = "user://bitacora_forense.txt"
	var archivo = FileAccess.open(ruta, FileAccess.WRITE)
	if archivo:
		archivo.store_string(bitacora_texto)
		archivo.close()
		print("Contenido de bit√°cora:", bitacora_texto)

		print("Bit√°cora guardada en:", ruta)
	else:
		print("No se pudo guardar la bit√°cora.")

func verificar_fin_juego():
	for e in evidencias:
		if e.estado != EstadoEvidencia.RECOLECTADO:
			return
		# Todas recolectadas
	mostrar_resultados_finales()
	
func mostrar_resultados_finales():
	# Mostrar un panel con resultados
	print("üéâ Juego terminado")
	print("Puntaje final:", puntuacion)
	var aciertos = 0
	var fallos = 0
	# Supongamos que llevas contadores separados en GameManager
	print("Aciertos:", aciertos)
	print("Fallos:", fallos)
	# Aqu√≠ podr√≠as abrir un PopupPanel con esta informaci√≥n
